operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java

```diff
@@ -280,10 +255,65 @@ protected void assumeFirstClassCitizens(KeycloakStatusBuilder status) {
                 .map(ValueOrSecret::getName)
                 .collect(Collectors.toSet());
 
-        final var sameItems = CollectionUtil.intersection(serverConfigNames, FIRST_CLASS_FIELDS);
+        final var sameItems = CollectionUtil.intersection(serverConfigNames, firstClassConfigOptions);
         if (CollectionUtil.isNotEmpty(sameItems)) {
             status.addWarningMessage("You need to specify these fields as the first-class citizen of the CR: "
                     + CollectionUtil.join(sameItems, ","));
         }
     }
+
+    private <T> OptionMapper<T> optionMapper(T optionSpec) {
+        return new OptionMapper<>(optionSpec);
+    }
+
+    private class OptionMapper<T> {
+        private final T categorySpec;
+        private final List<EnvVar> envVars;
+
+        public OptionMapper(T optionSpec) {
+            this.categorySpec = optionSpec;
+
+            var kcContainer = deployment.getSpec().getTemplate().getSpec().getContainers().get(0);
+            var envVars = kcContainer.getEnv();
+            if (envVars == null) {
+                envVars = new ArrayList<>();
+                kcContainer.setEnv(envVars);
+            }
+            this.envVars = envVars;
+        }
+
+        public <R> OptionMapper<T> mapOption(String optionName, Function<T, R> optionValueSupplier) {
+            firstClassConfigOptions.add(optionName);
+
+            if (categorySpec == null) {
+                Log.debugf("No category spec provided for %s", optionName);
+                return this;
+            }
+
+            R value = optionValueSupplier.apply(categorySpec);
+            String valueStr = String.valueOf(value);
+
+            if (value == null || valueStr.trim().isEmpty()) {
+                Log.debugf("No value provided for %s", optionName);
+                return this;
+            }
+
+            EnvVar envVar = new EnvVarBuilder()
+                    .withName(getEnvVarName(optionName))
+                    .withValue(valueStr)
+                    .build();
+
+            envVars.add(envVar);
+
+            return this;
+        }
+
+        public <R> OptionMapper<T> mapOption(String optionName, R optionValue) {
+            return mapOption(optionName, s -> optionValue);
+        }
+
+        protected <R extends Collection<?>> OptionMapper<T> mapOptionFromCollection(String optionName, Function<T, R> optionValueSupplier) {
+            return mapOption(optionName, s -> optionValueSupplier.apply(s).stream().map(String::valueOf).collect(Collectors.joining(",")));
```

Comment:
```suggestion
            return mapOption(optionName, s -> {
                var value = optionValueSupplier.apply(s);
                if (value == null) return null;
                return value.stream().map(String::valueOf).collect(Collectors.joining(","));
            });
```
The NPE can be thrown here. For instance, when only the enabled features are specified, but not disabled ones. Maybe not so nice code, but I think it's pretty good readable, and intuitive. WDYT? 

Probably would be good to add some test cases for that kind of logic.

Reply:
Good catch, thanks! Commited your suggestion and added a test for it.